server/controllers/logger.ts

```diff
@@ -22,51 +22,55 @@ interface UserData {
   recommendation: string;
 }
 
-//func to insert and get data from the database
-export const insertUserData = async (
-  userData: UserData,
-): Promise<UserData | null> => {
-  const { data, error } = await supabase.from('user_data').insert([userData]);
+export const insertUserDataMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  const userData: UserData = res.locals;
 
-  if (error) {
-    console.error('Error inserting data:', error);
-    return null;
+  if (!userData) {
+    return next(error);
   }
-  console.log('Data inserted successfully:', data);
-  return data[0];
-};
 
-// Fetch all user data from the database
-export const getUserData = async (): Promise<UserData[] | null> => {
-  const { data, error } = await supabase.from('user_data').select('*');
+  try {
+    const { data, error } = await supabase.from('user_data').insert([userData]);
+
+    if (error) {
+      console.error('Error inserting data:', error);
+      return res.status(500).json({ error: 'Failed to insert data' });
+    }
 
-  if (error) {
-    console.error('Error retrieving data:', error);
-    return null;
+    console.log('Data inserted successfully:', data);
+    // Store inserted data in `res.locals` for further middleware
+    res.locals.insertedData = data[0];
+    next();
+  } catch (err) {
+    console.error('Error during insertion:', err);
+    res.status(500).json({ error: 'An unexpected error occurred' });
   }
-  return data;
 };
 
-//test on get user data
-// const getData = async () => {
-//   const data = await getUserData();
-//   console.log('Retrieved Data:', data);
-// };
-
-// getData();
+// Middleware to get all user data
+export const getUserDataMiddleware = async (
```

Comment:
@nanyi-deng I'm not sure that we need this functionality, as we will probably only manually pull data from our database? Not sure.